agent/doctor/docker_runtime_healthcheck.go

```diff
@@ -25,23 +25,32 @@ import (
 const systemPingTimeout = time.Second * 2
 
 type dockerRuntimeHealthcheck struct {
+	// HealthcheckType is the reported healthcheck type
+	HealthcheckType string `json:"HealthcheckType,omitempty"`
 	// Status is the container health status
-	Status HealthcheckStatus `json:"Status,omitempty"`
+	Status HealthcheckStatus `json:"HealthcheckStatus,omitempty"`
 	// Timestamp is the timestamp when container health status changed
 	TimeStamp time.Time `json:"TimeStamp,omitempty"`
+	// StatusChangeTime is the latest time the health status changed
+	StatusChangeTime time.Time `json:"StatusChangeTime,omitempty"`
+
 	// LastStatus is the last container health status
-	LastStatus HealthcheckStatus `json:"lastStatus,omitempty"`
+	LastStatus HealthcheckStatus `json:"LastStatus,omitempty"`
 	// LastTimeStamp is the timestamp of last container health status
 	LastTimeStamp time.Time `json:"LastTimeStamp,omitempty"`
-	client        dockerapi.DockerClient
-	lock          sync.RWMutex
+
+	client dockerapi.DockerClient
+	lock   sync.RWMutex
 }
 
 func NewDockerRuntimeHealthcheck(client dockerapi.DockerClient) *dockerRuntimeHealthcheck {
+	nowTime := time.Now()
 	return &dockerRuntimeHealthcheck{
-		client:    client,
-		Status:    HealthcheckStatusInitializing,
-		TimeStamp: time.Now(),
+		HealthcheckType:  HealthcheckTypeContainerRuntime,
+		Status:           HealthcheckStatusInitializing,
+		TimeStamp:        nowTime,
+		StatusChangeTime: nowTime,
```

Comment:
Should we set `StatusChangeTime` to nil the first time ?

Reply:
This is assuming that the initial status is a status change from nothing to HealthcheckStatusInitializing.  As a customer I would expect the status change time to be updated any time the status changes -- including the case of no status -> some status.